rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -653,81 +665,115 @@ static std::string get_node_user_data(const char * node_name, const char * node_
 {
-  rcutils_allocator_t allocator = rcutils_get_default_allocator();
-
-  char * security_file_path = get_security_file_URI(
-    file_name, security_options->security_root_path, allocator);
-  if (security_file_path != nullptr) {
-    dds_qset_prop(qos, qos_property_name, security_file_path);
-    allocator.deallocate(security_file_path, allocator.state);
-  }
+  allocator.deallocate(dds_security_files.identity_ca_cert, allocator.state);
+  dds_security_files.identity_ca_cert = nullptr;
+  allocator.deallocate(dds_security_files.cert, allocator.state);
+  dds_security_files.cert = nullptr;
+  allocator.deallocate(dds_security_files.key, allocator.state);
+  dds_security_files.key = nullptr;
+  allocator.deallocate(dds_security_files.permissions_p7s, allocator.state);
+  dds_security_files.permissions_p7s = nullptr;
+  allocator.deallocate(dds_security_files.governance_p7s, allocator.state);
+  dds_security_files.governance_p7s = nullptr;
+  allocator.deallocate(dds_security_files.permissions_p7s, allocator.state);
+  dds_security_files.permissions_p7s = nullptr;
 }
+
 #endif  /* RMW_SUPPORT_SECURITY */
 
-/*  Set all the qos properties needed to enable DDS security  */
+/* Attempt to set all the qos properties needed to enable DDS security */
 rmw_ret_t configure_qos_for_security(
   dds_qos_t * qos, const rmw_node_security_options_t * security_options)
 {
 #if RMW_SUPPORT_SECURITY
-  /*  File path is set to nullptr if file does not exist or is not readable  */
-  store_security_filepath_in_qos(
-    qos, "dds.sec.auth.identity_ca", "identity_ca.cert.pem",
-    security_options);
-  store_security_filepath_in_qos(
-    qos, "dds.sec.auth.identity_certificate", "cert.pem",
-    security_options);
-  store_security_filepath_in_qos(
-    qos, "dds.sec.auth.private_key", "key.pem",
-    security_options);
-  store_security_filepath_in_qos(
-    qos, "dds.sec.access.permissions_ca", "permissions_ca.cert.pem",
-    security_options);
-  store_security_filepath_in_qos(
-    qos, "dds.sec.access.governance", "governance.p7s",
-    security_options);
-  store_security_filepath_in_qos(
-    qos, "dds.sec.access.permissions", "permissions.p7s",
-    security_options);
-
-  dds_qset_prop(qos, "dds.sec.auth.library.path", "dds_security_auth");
-  dds_qset_prop(qos, "dds.sec.auth.library.init", "init_authentication");
-  dds_qset_prop(qos, "dds.sec.auth.library.finalize", "finalize_authentication");
-
-  dds_qset_prop(qos, "dds.sec.crypto.library.path", "dds_security_crypto");
-  dds_qset_prop(qos, "dds.sec.crypto.library.init", "init_crypto");
-  dds_qset_prop(qos, "dds.sec.crypto.library.finalize", "finalize_crypto");
-
-  dds_qset_prop(qos, "dds.sec.access.library.path", "dds_security_ac");
-  dds_qset_prop(qos, "dds.sec.access.library.init", "init_access_control");
-  dds_qset_prop(qos, "dds.sec.access.library.finalize", "finalize_access_control");
+  bool qos_configured = false;
```

Comment:
You could make this a `rmw_ret_t ret = RMW_RET_UNSUPPORTED;` and just `return ret;` at the end instead of carrying a boolean that does the same thing.

Reply:
Updated in bf8ccfa